### PR TITLE
Add M110 release items

### PR DESCRIPTION
**Pull request by 1c3t3a**
This adds support for all items released in M110.

### CODE REVIEW

File: src/__tests__/dpsl.test.js [hunk 1]
```diff
@@ -510,4 +530,45 @@ describe('dpsl.diagnostics tests', () => {
...
+  });
+
+  test('runSmartctlCheckRoutine works with parameters', (done) => {
+    const expectedRunRoutineResponse = {id: 123456};
+
+    const chrome = {
+      os: {
+        diagnostics: {
+          runSmartctlCheckRoutine:
+            (params = undefined) => Promise.resolve(expectedRunRoutineResponse),
```
Comment (Lamzin) [inline src/__tests__/dpsl.test.js:561]: Maybe here we should resolve promise only if params is expected?

File: src/diagnostics_manager.js [hunk 2]
```diff
@@ -305,23 +305,65 @@ class DiskManager {
...
-  async runSmartctlCheckRoutine() {
+  async runSmartctlCheckRoutine(params = undefined) {
     const functionName = 'runSmartctlCheckRoutine';
     if (!isSupported(functionName)) {
       throw new MethodNotFoundError(API_NAME, functionName,
           /* chromeVersion */ 100);
     }
 
-    return chrome.os.diagnostics.runSmartctlCheckRoutine().then(
+    return chrome.os.diagnostics.runSmartctlCheckRoutine(params).then(
```
Comment (Lamzin) [inline src/diagnostics_manager.js:349]: will it work if we pass undefined on old versions of browser?

maybe we should call runSmartctlCheckRoutine() without any arguments if params is undefined?

**Review by Lamzin — COMMENTED** [review]
Thanks Bastian for the PR!

I have few comments.

File: src/__tests__/dpsl.test.js (hunk 1 above)
Comment (1c3t3a) [reply src/__tests__/dpsl.test.js:561]: Done! :)

File: src/diagnostics_manager.js (hunk 2 above)
Comment (1c3t3a) [reply src/diagnostics_manager.js:349]: Good call, that makes sense!

File: src/__tests__/dpsl.test.js [hunk 3]
```diff
@@ -510,4 +530,47 @@ describe('dpsl.diagnostics tests', () => {
...
+
+  test('runSmartctlCheckRoutine works without parameters', (done) => {
+    const expectedRunRoutineResponse = {id: 123456};
+
+    const chrome = {
+      os: {
+        diagnostics: {
+          runSmartctlCheckRoutine:
+             (params = undefined) =>
+               Promise.resolve(expectedRunRoutineResponse),
```
Comment (Lamzin) [inline src/__tests__/dpsl.test.js:543]: nit: i'd resolve this only when params is undefined

Comment (Lamzin) [reply src/__tests__/dpsl.test.js:543]: Looks like we actually can check whether arguments.length === 0 to verify that there were no input params

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

File: src/__tests__/dpsl.test.js [hunk 4]
```diff
@@ -510,4 +530,47 @@ describe('dpsl.diagnostics tests', () => {
...
+
+  test('runSmartctlCheckRoutine works with parameters', (done) => {
+    const expectedRunRoutineResponse = {id: 123456};
+
+    const chrome = {
+      os: {
+        diagnostics: {
+          runSmartctlCheckRoutine:
+             (params = undefined) =>
+               Promise.resolve(params ? expectedRunRoutineResponse : undefined),
```
Comment (Lamzin) [inline src/__tests__/dpsl.test.js:563]: Same here: we can use arguments to check that params was provided and we can validate that params. percentageUsedThreshold is equal to 42 before resolving the promise

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments

File: src/diagnostics_manager.js [hunk 5]
```diff
@@ -305,26 +305,73 @@ class DiskManager {
...
    */
-  async runSmartctlCheckRoutine() {
+  async runSmartctlCheckRoutine(params = undefined) {
     const functionName = 'runSmartctlCheckRoutine';
     if (!isSupported(functionName)) {
       throw new MethodNotFoundError(API_NAME, functionName,
           /* chromeVersion */ 100);
     }
 
+    if (params) {
```
Comment (Lamzin) [inline src/diagnostics_manager.js:349]: same: can we use agruments here?

File: src/__tests__/dpsl.test.js (hunk 3 above)
Comment (1c3t3a) [reply src/__tests__/dpsl.test.js:543]: Seems like the [jest](https://github.com/facebook/jest) framework we use for testing injects some parameters to the function call, so we can't use `arguments` in test. I sticked with `undefined` as it does the job and used `arguments` in the Diagnostics manager.

**Review by Lamzin — APPROVED** [review]

**Pull request merged**